core/keystore/src/lib.rs

```diff
@@ -151,22 +145,123 @@ impl Store {
 		Ok(public_keys)
 	}
 
-	fn key_file_path<TPair: Pair + TypedKey>(&self, public: &TPair::Public) -> PathBuf {
+	fn key_file_path<TPair: Pair>(&self, public: &TPair::Public) -> PathBuf {
 		let mut buf = self.path.clone();
 		let bytes: [u8; 4] = TPair::KEY_TYPE.to_le_bytes();
 		let key_type = hex::encode(bytes);
 		let key = hex::encode(public.as_slice());
 		buf.push(key_type + key.as_str());
 		buf
 	}
+
+	/// Register a `StoreHandler`.
+	fn register_handler<TKey: TypedKey, THandler: StoreHandler + 'static>(
+		&mut self,
+		handler: THandler,
+	) {
+		self.handlers.push((TKey::KEY_TYPE, Box::new(handler)));
+	}
+}
+
+/// A local session key store supporting a single crypto algorithm.
+#[derive(Clone)]
+pub struct SessionStore<TPair: Pair> {
```

Comment:
why not a store supporting _all_ crypto algorithms?

Reply:
the idea is that aura/babe and grandpa get their own session store and only subscribe to the crypto they want. and if libp2p needs their secio key it also would get it's own session store. Do you disagree with the design?

Reply:
The disadvantage of supporting multiple cryptos concurrently is that we have to store them as bytes and then decode/encode every time.